### PR TITLE
Gpl 969 test data generation

### PR DESCRIPTION
**Pull request by harrietc52**

### CODE REVIEW

File: pages/uat_actions/generate_test_run.vue [hunk 1]
```diff
@@ -6,7 +6,19 @@
...
+        Select the number of plates you want to generate test data for with the number of positive samples per plate.
+        You can add additional sets of plates up to a maximum of {{ maxNumberOfPlates }}.
+      </b-card-text>
+      <b-card-text>
+        N.B. for Beckman testing remember to check the
+        <em>Add to DART</em> box. DART data is not needed for Biosero testing.
+      </b-card-text>
+      <b-card-text>
+        The
+        <em>Generate</em> button will start the creation of the data, and you will be automatically taken to the label printing screen when it is done.
```
Comment (sdjmchattie) [inline pages/uat_actions/generate_test_run.vue:20]: Some odd word wrapping on these two lines, but this text is great!

File: pages/uat_actions/generate_test_run.vue [hunk 2]
```diff
@@ -91,22 +125,36 @@ export default {
...
     }
   },
   methods: {
     numberOfPositivesOptions() {
-      return Array.from({length: MAX_NUMBER_OF_POSITIVES + 1}, (v, i) => i) // 0 - 96
+      return Array.from({length: this.maxNumberOfPositives + 1}, (v, i) => i) // 0 - 96
     },
     numberOfPlatesOptions() {
-      return Array.from({length: MAX_NUMBER_OF_PLATES}, (v, i) => i + 1) // 1 - 200
+      return Array.from({length: this.maxNumberOfPlates - this.totalPlates}, (v, i) => i + 1) // 200 - totalPlates
```
Comment (sdjmchattie) [inline pages/uat_actions/generate_test_run.vue:153]: I'm intrigued what happens to the drop down when you add exactly 200 plates?  It becomes a drop down with no options in it?  I see you disable the "Add" button in this scenario, but maybe the whole add row should disappear when you hit 200?

**Review by sdjmchattie — CHANGES_REQUESTED** [review]

File: pages/uat_actions/generate_test_run.vue (hunk 2 above)
Comment (harrietc52) [reply pages/uat_actions/generate_test_run.vue]: Yes, it had left an empty drop down so I had made the 200/200 turn red. But think your right - just added a v-if, to only show the drop down if total isn't reached 

Screenshot 2021-07-21 at 15 16 54

File: pages/uat_actions/generate_test_run.vue (hunk 1 above)
Comment (harrietc52) [reply pages/uat_actions/generate_test_run.vue:20]: Some auto formatting thing :(

File: pages/uat_actions/generate_test_run.vue [hunk 3]
```diff
@@ -16,53 +29,77 @@
...
-        <b-button id="addButton" variant="outline-primary" @click="add">Add</b-button>
+        <b-button
+          id="addButton"
+          variant="outline-primary"
+          :disabled="isBusy || totalPlates>=maxNumberOfPlates"
+          @click="add"
+        >Add</b-button>
       </b-form>
 
+      <b-card v-else-if="totalPlatesReached">
```
Comment (sdjmchattie) [inline pages/uat_actions/generate_test_run.vue:48]:
```suggestion
      <b-card v-else>
```

**Review by sdjmchattie — DISMISSED** [review]

**Review by andrewsparkes — APPROVED** [review]

**Review by sdjmchattie — APPROVED** [review]

**Pull request merged**